src/main/java/io/reactivex/internal/operators/completable/CompletableOnSubscribeConcatArray.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright 2015 Netflix, Inc.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is
+ * distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See
+ * the License for the specific language governing permissions and limitations under the License.
+ */
+
+package io.reactivex.internal.operators.completable;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import io.reactivex.Completable;
+import io.reactivex.Completable.*;
+import io.reactivex.disposables.*;
+
+public final class CompletableOnSubscribeConcatArray implements CompletableOnSubscribe {
```

Comment:
@akarnokd do you really see much sense in a separate class for arrays?

You can provide an override where passed array will be wrapped into `ArrayList` and then passed to the `CompletableOnSubscribeConcatIterable`.

Less code — less bugs :)